tests/unit/hook_packages/terraform/test_prepare_hook.py

```diff
@@ -961,6 +963,103 @@ def test_translate_to_cfn_with_mapping_s3_source_to_function(self, checksum_mock
         self.assertEqual(translated_cfn_dict, self.expected_cfn_with_child_modules_and_s3_source_mapping)
         mock_enrich_mapped_resources.assert_not_called()
 
+    @parameterized.expand(
+        [
+            ("expected_cfn_lambda_function_resource_zip", "tf_lambda_function_resource_zip_sam_metadata", "Zip"),
+            (
+                "expected_cfn_image_package_type_lambda_function_resource",
+                "tf_image_package_type_lambda_function_resource_sam_metadata",
+                "Image",
+            ),
+        ]
+    )
+    def test_validate_referenced_resource_matches_sam_metadata_type_valid_types(
```

Comment:
Wondering if we can write this test more explicitly... kind of like the opposite of `assertRaises`. But as there is not `assertNotRaises`, can we try something like ?

```
try:
  _validate_referenced_resource_matches_sam_metadata_type(...)
except InvalidSamMetadataProperties:
  self.fail("some reason")
```

Reply:
Done